### ISSUE TITLE
More complex example in spreadsheet collection

### ISSUE THREAD

**Issue by ldonnet** [westonganger/spreadsheet_architect#1]
Hi,

I use your gem for the first time and I try to use it to export in xlsx with specific needs but I could not find the right syntax : 

```ruby
  def self.spreadsheet_columns
    [
      [ "Date", "I18n.l(created_at.to_date, format: :long)" ],
      [ I18n.t('payments.csv.id'), :id ],
      [ I18n.t('payments.csv.login'), :payer_login ],
      [ I18n.t('payments.csv.email'), :payer_email ],
      [ I18n.t('payments.csv.firstname'), :payer_first_name ],
      [ I18n.t('payments.csv.lastname'), :payer_last_name ],
      [ I18n.t('payments.csv.type'), "type.slice(0, type.index('Payment'))" ],
      [ I18n.t('payments.csv.amount'), "#{amount} #{currency.symbol}" ],
      [ I18n.t('payments.csv.reward'), 'reward.present? ? (reward.amount - reward.description) : I18n.t("payments.csv.no_rewards")' ],
      [ I18n.t('payments.csv.address'), 'try(:payer_location).try(:address_complete) || "-"' ]
    ]
  end
```

I have go just 2 errors on these lines : 
```ruby
      [ I18n.t('payments.csv.amount'), "#{amount} #{currency.symbol}" ],
      [ I18n.t('payments.csv.reward'), 'reward.present? ? (reward.amount - reward.description) : I18n.t("payments.csv.no_rewards")' ],
```

Is there a way to do this with your gem?

**Comment by westonganger** [comment]
Okay so you cant do `"#{amount}  #{currency.symbol}"` this is because the method is defined on the class which does not have the method `amount` or `currency`, that method is on the instance. 

For the first one, can you try to write it like this: `'"#{amount} #{currency.symbol}"'` (notice the single quotes on the outside.)

For the second one, can you try to write it like this: `'"#{reward.present? ? (reward.amount - reward.description) : I18n.t("payments.csv.no_rewards")}"'` (notice the single quotes on the outside.)

The alternative to writing them like this is that you define a method on the instance

```ruby
def the_amount
  "#{amount} #{currency_symbol}"
end

def the_reward
  reward.present? ? (reward.amount - reward.description) : I18n.t("payments.csv.no_rewards")
end

def self.spreadsheet_columns
  [
    [ "Date", "I18n.l(created_at.to_date, format: :long)" ],
    [ I18n.t('payments.csv.id'), :id ],
    [ I18n.t('payments.csv.login'), :payer_login ],
    [ I18n.t('payments.csv.email'), :payer_email ],
    [ I18n.t('payments.csv.firstname'), :payer_first_name ],
    [ I18n.t('payments.csv.lastname'), :payer_last_name ],
    [ I18n.t('payments.csv.type'), "type.slice(0, type.index('Payment'))" ],
    [ I18n.t('payments.csv.amount'), :the_amount],
    [ I18n.t('payments.csv.reward'), :the_reward],
    [ I18n.t('payments.csv.address'), 'try(:payer_location).try(:address_complete) || "-"' ]
  ]
end
```

Im going to look into the `spreadsheet_columns` method to see if it can be moved to the instance instead of the class possibly.

**Comment by ldonnet** [comment]
Yeah you are awesome it works!!

I have just a strange behaviour in excel (last version) contents in cell are underlined. Perhaps I need to specify specific style? Or a wrong default stylesheet?

Thanks for your great work and your time!

**Comment by westonganger** [comment]
I am releasing 1.0.5 after I deal with this underline issue. Do you think you could try it out using the options below to see if that works properly? If this works I will fix this in 1.0.5

```ruby
.to_xlsx(header_style: {bold: :none}, row_style: {bold: :none}) 
# and
.to_xlsx(header_style: {bold: :single}, row_style: {bold: :single})
```

**Comment by westonganger** [comment]
If you want to have different info per controller you can utilize the `spreadsheet_columns` option when you call `to_xlsx` . Though in 1.0.5 you must only use symbols methods when using this option.

**Comment by westonganger** [comment]
I just released version 1.1.0 (which i was previously referring to 1.0.5). The spreadsheet_columns method has been moved from the class to the instance. See the Model section of the readme for new usage instructions. Also the underline issue should be fixed and should not require the underline option anymore and should work correctly with true/false
